pulp_service/pulp_service/app/serializers.py

```diff
@@ -63,12 +67,40 @@ class ContentScanSerializer(serializers.Serializer, ValidateFieldsMixin):
 
     def validate(self, data):
         data = super().validate(data)
-        if bool(data.get("repo_version", None)) == bool(data.get("package_json", None)):
+        repo_ver = data.get("repo_version", None)
+        pkg_json = data.get("package_json", None)
+        if bool(repo_ver) == bool(pkg_json):
             raise serializers.ValidationError(
                 _("Exactly one of 'repo_version' or 'package_json' must be specified.")
             )
+
+        # no more validations needed for pkg_json
+        if pkg_json:
+            return data
+
+        # for rpm repositories we need to verify the repository labels
+        if repo_ver.repository.pulp_type == "rpm.rpm":
+            if not self._validate_rpm_repo_expected_fields(repo_ver.repository):
+                raise serializers.ValidationError(
+                    _("Repository ecosystem not supported or does not contain the expected labels.")
+                )
+
         return data
 
+    def _validate_rpm_repo_expected_fields(self, repo):
+        """
+        verify if the label 'osv.dev ecosystem' == 'Red Hat'
+        verify if len(label['osv.dev cpe']) > 0
+        """
+        # for now, we are only supporting 'Red Hat' for rpms
+        if repo.pulp_labels.get(OSV_RH_ECOSYSTEM_LABEL) == "Red Hat":
+            # 'osv.dev cpe' must be provided for Red Hat ecosystem
+            return (
+                repo.pulp_labels.get(OSV_RH_ECOSYSTEM_CPES_LABEL)
```

Comment:
You could maybe make this simpler+faster with the walrus also. It's a stylistic choice. Up to you.